scripts/form.js

```diff
@@ -0,0 +1,34 @@
+const submitFinal = document.getElementById('buttonFinal');
+const submitButton = document.createElement('button')
+submitButton.setAttribute('type', 'submit');
+submitButton.textContent = 'Submit';
+submitFinal.appendChild(submitButton);
+
+let total = 0;
+
+$(document).ready(function () {
+  $('#buttonFinal').click(function myAdoption() {
+  const form = document.getElementById('buttonFinal');
+  form.addEventListener('submit', function(event) {
```

Comment:
A submit event should be attached to the actual form element and not the button element. (by the way your #buttonFinal is actually targeting a div and not the button itself)
Buttons never trigger a submit event, so this code is never running.